File: TravelApp/TravelApp.Models/Trip.cs
using System;
using System.ComponentModel.DataAnnotations;
using TravelApp.Common.BaseModels;

namespace TravelApp.Models
{
    public class Trip : BaseDeletableModel<string>, IAuditInfo
    {
        public Trip()
        {
            this.Id = Guid.NewGuid().ToString();
        }

        public string CurrentLocation { get; set; }

        public Order Order { get; set; }

        public string OrderId { get; set; }

        public ApplicationUser ApplicationUser { get; set; }

        public string ApplicationUserId { get; set; }

        public string Status { get; set; }
    }
}

File: TravelApp/TravelApp/ClientApp/src/_models/trip.ts
import { Order } from "./order";
import { User } from "./user";

export class Trip {
    id: string;
    orderId: string;
    driverId: string;
    status: string;
    createdOn: Date;
    isCompleted: boolean;
    order: Order;
    applicationUser: User;
}